Chatkit/Transformer/Transformer.swift

```diff
@@ -0,0 +1,115 @@
+
+protocol Transformer {
+    
+    func transform(state: RoomState) -> Room
+    func transform(currentState: VersionedState, previousState: VersionedState?) -> JoinedRoomsRepository.ChangeReason?
```

Comment:
I do agree that this will have to be refactored in future as we will have more transform methods. However, I am not sure if we should rename methods to be more descriptive or perhaps we should split the code into a few more structures.

Reply:
OK, leave as is for now.  👍